### DISCUSSION TITLE
CancelableRequest.json() returns CancelableRequest and not Promise

### DISCUSSION THREAD

**Discussion by Inrixia** [sindresorhus/got#2141] [General]
#### What would you like to discuss?

I'm in the process of updating to the latest version of got. According to the docs [CancelableRequest.json() returns a Promise](https://github.com/sindresorhus/got/blob/main/documentation/1-promise.md#promisejsont)

However the types specified for got show it returning a CancelableRequest which means I just can't get a proper object out of it???
https://github.com/sindresorhus/got/blob/f0ac0b3fdfa82f53edfdc821ae3afa398b1bab0a/source/as-promise/types.ts#L36

Honestly the whole change from promises to CancelableRequests is rather confusing, most of the docs refer to got returning a Promise while the new types all use `CancelableRequest` which doesn't support any of the usual Promise interface... Is there something I'm missing?

#### Checklist

- [x] I have read the documentation.

**Comment by sindresorhus** [comment]
> According to the docs [CancelableRequest.json() returns a Promise](https://github.com/sindresorhus/got/blob/main/documentation/1-promise.md#promisejsont)

It technically does. `CancelableRequest` extends `PCancelable` which extends `Promise`.

**Reply by Inrixia** [reply]
According to typescript awaiting it does not give the actual value though so something isn't working

**Reply by Inrixia** [reply]
This is also reinforced by the fact I cannot use .then() with it

**Reply by Inrixia** [reply]
So any ideas on why?

**Reply by Inrixia** [reply]
@sindresorhus any help?

**Reply by szmarczak** [reply]
Most likely it's an issue with TypeScript on your end or either it's a bug in TypeScript language server.

**Reply by Inrixia** [reply]
Thanks, checked through and it was a sneaky typing issue due to capitalization in one of the core files. After fixing it typing is working correctly now

**Reply by cmjacques** [reply]
Hi @Inrixia - can you explain your solution a bit further? I've run into the same thing. Thanks.

**Reply by Inrixia** [reply]
@cmjacques sure. I beleive the capitalization between versions has changed. You can see a working example in this repo https://github.com/Inrixia/floatplaneAPI.ts/blob/main/src/index.ts
